api/middleware/roleTicketsMiddleware.js

```diff
@@ -0,0 +1,36 @@
+const RoleTicket = require('../roleTickets/roleTicketsModel');
+const roleTicketSchema = require('../../data/schemas/roleTicketSchema');
+
+async function checkRoleTicketIdExists(req, res, next) {
+  const { role_ticket_id } = req.params;
+  const badIdErr = {
+    status: 404,
+    message: `Role Ticket with ID ${role_ticket_id} not found!`,
+  };
+  try {
+    const roleTicket = await RoleTicket.findByRoleTicketById(role_ticket_id);
+    if (!roleTicket) {
+      next(badIdErr);
```

Comment:
I would explicitly return `next()` here; in theory it shouldn't make a difference, but I've witnessed code _after_ calls to `next()` getting run when `next()` is not "returned." Might've just been an oddball moment, but better safe than sorry. :-)